server/src/main/java/org/opensearch/indices/RemoteStoreSettings.java

```diff
@@ -105,9 +105,21 @@ public class RemoteStoreSettings {
         Property.NodeScope
     );
 
+    /**
+     * Controls timeout value while uploading segment files to remote segment store
+     */
+    public static final Setting<TimeValue> CLUSTER_REMOTE_SEGMENT_TRANSFER_TIMEOUT_SETTING = Setting.timeSetting(
+        "cluster.remote_store.segment.transfer_timeout",
+        TimeValue.timeValueHours(3),
```

Comment:
Curious to know how we decided this time-out limit?

Reply:
We can keep it lower I feel, maybe 30 mins